solr/core/src/java/org/apache/solr/core/SolrCores.java

```diff
@@ -36,60 +36,46 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-class SolrCores {
+/** Holds/manages {@link SolrCore}s within {@link CoreContainer}. */
+public class SolrCores {
+
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  // for locking around manipulating any of the core maps.
+  protected final Object modifyLock = new Object();
 
-  private static final Object modifyLock =
-      new Object(); // for locking around manipulating any of the core maps.
   private final Map<String, SolrCore> cores = new LinkedHashMap<>(); // For "permanent" cores
 
   // These descriptors, once loaded, will _not_ be unloaded, i.e. they are not "transient".
   private final Map<String, CoreDescriptor> residentDescriptors = new LinkedHashMap<>();
 
   private final CoreContainer container;
 
-  private Set<String> currentlyLoadingCores =
+  private final Set<String> currentlyLoadingCores =
       Collections.newSetFromMap(new ConcurrentHashMap<String, Boolean>());
 
-  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
-
   // This map will hold objects that are being currently operated on. The core (value) may be null
   // in the case of initial load. The rule is, never to any operation on a core that is currently
   // being operated upon.
-  private static final Set<String> pendingCoreOps = new HashSet<>();
+  private final Set<String> pendingCoreOps = new HashSet<>();
 
   // Due to the fact that closes happen potentially whenever anything is _added_ to the transient
   // core list, we need to essentially queue them up to be handled via pendingCoreOps.
-  private static final List<SolrCore> pendingCloses = new ArrayList<>();
```

Comment:
a couple fields in here like modifyLock and this were static which made no sense so I removed that.